app/presenters/menu/manager.rb

```diff
@@ -1,11 +1,17 @@
 module Menu
   class Manager
+    include Enumerable
     include Singleton
 
     class << self
       extend Forwardable
 
-      delegate %i(menu item_in_section? item section section_id_string_to_symbol each) => :instance
+      delegate %i(menu item_in_section? item section section_id_string_to_symbol
+                  each map detect select) => :instance
+    end
+
+    def each
+      @menu.map { |section| yield section }
```

Comment:
Uh, maybe I'm not following this enough, but having a `map` instead of `each` is definitely evil. Even if you can convince me that this is *The Right Way&trade;*, please add some comments here.
![evilest](https://cloud.githubusercontent.com/assets/649130/24402821/8182bc42-13ba-11e7-9cd0-b013814002d4.gif)